light-client/src/tests.rs

```diff
@@ -152,25 +152,42 @@ pub fn verify_single(
     clock_drift: Duration,
     now: Time,
 ) -> Result<LightBlock, Verdict> {
+    trait IntoResult<T, E> {
+        fn into_result(self, ok: T) -> Result<T, E>;
+    }
+
+    impl<T> IntoResult<T, Verdict> for Verdict {
+        fn into_result(self, ok: T) -> Result<T, Verdict> {
+            match self {
+                Verdict::Success => Ok(ok),
+                error => Err(error),
+            }
+        }
+    }
```

Comment:
Nit: the purpose of this could look more obvious if this were just a helper function. Then again, method chaining is neat.

Reply:
As an aside, not related to this PR: why did there have to be a non-idiomatic `Verdict`, rather than a `Result<(), VerificationError>` where `VerificationError` would be an enum or otherwise easily matchable to that variant?
Basically, what the existing `impl From<Result<(), VerificationError>> for Verdict` does could be just how the `Result` returned value is supposed to be used if you care about the `NotEnoughTrust` case. Cc @romac

Reply:
> Agreed, let's do that.

I didn't mean this to be done in scope for this PR, which so far has not introduced breaking changes, but if you think it would be good and easy thing to do in the next release, I'm all for it.

Reply:
Indeed, let's target this for 0.27.